### ISSUE TITLE
Numbers reported in 'counts' files

### ISSUE THREAD

**Issue by mderidder95** [darwin-eu-dev/TreatmentPatterns#196]
Which counts are presented in files countsSex, countsAge, countsYear? The sums of counts in these files are equal, but these are not the sum of paths unstratified (all-all-all), neighter with or without path "None".

**Comment by mvankessel-EMC** [comment]
Hi Maria, could you share some dummy data? It is possible that there might be some censoring going on. As an example, if the minimum frequency is set to 5, age groups with fewer than 5 persons will be censored with the label `<5`. So if you compare both totals, there would indeed be a difference.

**Comment by mderidder95** [comment]
I did analyses with minFreq = 1. Example:
```
library(dplyr)

tempDir <- getwd()

con <- DBI::dbConnect(duckdb::duckdb(), CDMConnector::eunomia_dir())
cdm <- CDMConnector::cdm_from_con(
  con = con,
  cdm_schema = "main",
  write_schema = "main"
)

# Use the json files in TreatmentPatterns package to create cohorts
exampleCohortSet <- CDMConnector::readCohortSet(
  path = system.file(package = "TreatmentPatterns", "ExampleCohorts")
)
cohortTableName <- "cohort_table"
cdm <- CDMConnector::generateCohortSet(cdm, exampleCohortSet, name = cohortTableName)

# To get a clear situation, all treatments with duration below 30 days are removed

cdm$cohort_table <- cdm$cohort_table %>%
  mutate(duration = cohort_end_date - cohort_start_date)%>%
  filter(duration >= 30) %>%
  select(-duration)

cdm$cohort_table %>%
  group_by(cohort_definition_id) %>% count()
# Cohort 5 Death is empty.
# Cohort 6 and 7 also empty
# Cohort 8 is the target cohort. n = 2159.

cohorts <- exampleCohortSet %>%
  rename(cohortId = cohort_definition_id,
         cohortName = cohort_name) %>%
  mutate(type = c("event", "event", "event", "event", "exit", "event", "event", "target")) %>%
  select(cohortId, cohortName, type)

cdmDir <- file.path(tempDir, "Example_Durations30_minFreq1")
TreatmentPatterns::executeTreatmentPatterns(
  cohorts = cohorts,
  cohortTableName = cohortTableName,
  outputPath = cdmDir,
  cdm = cdm,
  eraCollapseSize = 30,
  combinationWindow = 30,
  minPostCombinationDuration = 1,
  minEraDuration = 30,
  minFreq = 1
)
treatmentPathways <- read.csv(file.path(cdmDir, "treatmentPathways.csv"))
# Number 557 is reported several times

treatmentPathways %>%
  filter(sex == "all") %>%
  filter(age == "all") %>%
  filter(indexYear == "all") %>%
  filter(path != "None")%>% select(freq) %>%sum() # 515
treatmentPathways  %>% filter(path == "None") %>% select(freq) %>%sum() # 1644
# The sum of these is 2159.
# This is the number in target cohort

countsSex <- read.csv(file.path(cdmDir, "countsSex.csv"))
countsSex %>% select(n) %>%sum() # 557
countsAge <- read.csv(file.path(cdmDir, "countsAge.csv"))
countsAge %>% select(n) %>%sum() # 557
countsYear<- read.csv(file.path(cdmDir, "countsYear.csv"))
countsYear%>% select(n) %>%sum() # 557

```

**Comment by mvankessel-EMC** [comment]
Hi Maria, thank you for the example code. With the latest version of TreatmentPatterns (2.6.1), I get a count of 2200 subjects the treatmentPathways.csv, countsYear.csv, countsAge.csv, and countsSex.csv files. See the reprex below.

Could you update to the latest version, and try again?

## Without filtering
``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(TreatmentPatterns)
library(CDMConnector)

tempDir <- tempdir()

con <- DBI::dbConnect(
  drv = duckdb::duckdb(),
  CDMConnector::eunomia_dir()
)

cdm <- CDMConnector::cdm_from_con(
  con = con,
  cdm_schema = "main",
  write_schema = "main"
)

cohortSet <- CDMConnector::readCohortSet(
  path = system.file(package = "TreatmentPatterns", "ExampleCohorts")
)

cdm <- generateCohortSet(
  cdm = cdm,
  cohortSet = cohortSet,
  name = "cohort_table"
)

cohorts <- cohortSet %>%
  rename(
    cohortId = "cohort_definition_id",
    cohortName = "cohort_name"
  ) %>%
  mutate(
    type = c("event", "event", "event", "event", "exit", "event", "event", "target")) %>%
  select("cohortId", "cohortName", "type")

andromeda <- computePathways(
  cohorts = cohorts,
  cohortTableName = "cohort_table",
  cdm = cdm
)
#> 
#> Construct treatment pathways, this may take a while for larger datasets.
#> Original number of rows: 8352
#> After eraCollapseSize: 0
#> Selected 1544 
#> out of 8352 rows
#> Iteration: 1
#> Switches: 8352
#> FRFS Combinations: 4
#> LRFS Combinations: 1527
#> Selected 4 
#> out of 559 rows
#> Iteration: 2
#> Switches: 559
#> FRFS Combinations: 0
#> LRFS Combinations: 4
#> After combinationWindow: 555
#> Time needed to execute combination window 0.0701635837554932
#> Order the combinations.
#> After filterTreatments: 554
#> Adding drug sequence number.
#> After maxPathLength: 554
#> Adding concept names.
#> Ordering the combinations.
#> constructPathways done.

export(
  andromeda = andromeda,
  outputPath = tempDir,
  minCellCount = 1
)
#> Writing metadata to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/metadata.csv
#> Writing treatmentPathways to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/treatmentPathways.csv
#> Censoring 0 pathways with a frequency <1 to 1.
#> Writing summaryStatsTherapyDuraion to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/summaryStatsTherapyDuraion.csv
#> Writing countsYearPath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/countsYear.csv
#> Writing countsAgePath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/countsAge.csv
#> Writing countsSexPath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpcrzZ7t/countsSex.csv

treatmentPathways <- read.csv(file.path(tempDir, "treatmentPathways.csv"))
countsYear <- read.csv(file.path(tempDir, "countsYear.csv"))
countsAge <- read.csv(file.path(tempDir, "countsAge.csv"))
countsSex <- read.csv(file.path(tempDir, "countsSex.csv"))

treatmentPathways %>%
  filter(
    age == "all",
    sex == "all",
    indexYear == "all"
  ) %>%
  summarize(n = sum(freq)) %>%
  pull()
#> [1] 2200

sum(countsYear$n)
#> [1] 2200
sum(countsAge$n)
#> [1] 2200
sum(countsSex$n)
#> [1] 2200
```

<sup>Created on 2023-12-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mderidder95** [comment]
Okay. But this number (2200) is larger than the number in the target cohort (2159). So I still would like to know what is counted in these counts-files.

**Comment by mvankessel-EMC** [comment]
I have resolved this on dev: 61dd305012be5e3da00c7a5110861d5eead3244a

``` r
# remotes::install_github("darwin-eu-dev/TreatmentPatterns@dev")

library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(TreatmentPatterns)
library(CDMConnector)

tempDir <- tempdir()

con <- DBI::dbConnect(
  drv = duckdb::duckdb(),
  CDMConnector::eunomia_dir()
)

cdm <- CDMConnector::cdm_from_con(
  con = con,
  cdm_schema = "main",
  write_schema = "main"
)

cohortSet <- CDMConnector::readCohortSet(
  path = system.file(package = "TreatmentPatterns", "ExampleCohorts")
)

cdm <- generateCohortSet(
  cdm = cdm,
  cohortSet = cohortSet,
  name = "cohort_table"
)

cohortCount(cdm$cohort_table)
#> # A tibble: 8 × 3
#>   cohort_definition_id number_records number_subjects
#>                  <int>          <dbl>           <dbl>
#> 1                    1           2679            2679
#> 2                    2           2130            2130
#> 3                    3           1927            1927
#> 4                    4           2021            2021
#> 5                    6           1393            1393
#> 6                    7           1732            1732
#> 7                    8           2159            2159
#> 8                    5              0               0

cohorts <- cohortSet %>%
  rename(
    cohortId = "cohort_definition_id",
    cohortName = "cohort_name"
  ) %>%
  mutate(
    type = c("event", "event", "event", "event", "exit", "event", "event", "target")) %>%
  select("cohortId", "cohortName", "type")

andromeda <- computePathways(
  cohorts = cohorts,
  cohortTableName = "cohort_table",
  cdm = cdm
)
#> 
#> Construct treatment pathways, this may take a while for larger datasets.
#> Original number of rows: 8352
#> After eraCollapseSize: 0
#> Selected 1544 
#> out of 8352 rows
#> Iteration: 1
#> Switches: 8352
#> FRFS Combinations: 4
#> LRFS Combinations: 1527
#> Selected 4 
#> out of 559 rows
#> Iteration: 2
#> Switches: 559
#> FRFS Combinations: 0
#> LRFS Combinations: 4
#> After combinationWindow: 555
#> Time needed to execute combination window 0.0746112505594889
#> Order the combinations.
#> After filterTreatments: 554
#> Adding drug sequence number.
#> After maxPathLength: 554
#> Adding concept names.
#> Ordering the combinations.
#> constructPathways done.

export(
  andromeda = andromeda,
  outputPath = tempDir,
  minCellCount = 1
)
#> Writing metadata to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/metadata.csv
#> Writing treatmentPathways to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/treatmentPathways.csv
#> Censoring 0 pathways with a frequency <1 to 1.
#> Writing summaryStatsTherapyDuration to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/summaryStatsTherapyDuration.csv
#> Writing countsYearPath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/countsYear.csv
#> Writing countsAgePath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/countsAge.csv
#> Writing countsSexPath to C:\Users\MVANKE~1\AppData\Local\Temp\RtmpmKD0PW/countsSex.csv

treatmentPathways <- read.csv(file.path(tempDir, "treatmentPathways.csv"))

countsYear <- read.csv(file.path(tempDir, "countsYear.csv"))
countsAge <- read.csv(file.path(tempDir, "countsAge.csv"))
countsSex <- read.csv(file.path(tempDir, "countsSex.csv"))

treatmentPathways %>%
  filter(
    age == "all",
    sex == "all",
    indexYear == "all"
  ) %>%
  summarize(sum = sum(freq)) %>%
  pull()
#> [1] 2159

sum(countsYear$n)
#> [1] 2159
sum(countsAge$n)
#> [1] 2159
sum(countsSex$n)
#> [1] 2159
```

<sup>Created on 2023-12-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mvankessel-EMC** [comment]
Added into latest CRAN release (2.6.2)
